### PR TITLE
[SYCL] Add Panther Lake H & U to sycl_ext_oneapi_device_architecture

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch adds two new GPU architectures - Panther Lake H and Panther Lake U to sycl_ext_device_architecture extension.

Some small required NFC refactoring was also performed.

### CODE REVIEW

File: llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td [hunk 1]
```diff
@@ -188,8 +188,14 @@ class IntelTargetInfo<string Name, list<Aspect> Aspects, list<int> subGroupSizes
 : TargetInfo<Name, IntelBaseAspects # Aspects, subGroupSizesList>;
 // Note: only the "canonical" target names are listed here - see
 // SYCL::gen::resolveGenDevice().
-def : IntelTargetInfo<"intel_gpu_bmg_g21", Fp16Fp64Atomic64, Sg16_32>;
+//
+// TODO: instructions on how to get "???" values below are required. Currently
```
Comment (dm-vodopyanov) [inline llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td:192]: @jzc can you please add instructions here? If they are too big, they can be moved to sycl/doc/developer folder and referenced here.

**Review by gmlueck — APPROVED** [review]
spec changes LGTM

File: llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td (hunk 1 above)
Comment (AlexeySachkov) [reply llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td:192]: The easiest way is to run `sycl-ls --verbose`, but that requires you having the HW. Until you know the data, you can always register a target as _not_ supporting anything.

**Review by mdtoguchi — APPROVED** [review]

File: sycl/include/sycl/ext/oneapi/experimental/device_architecture.def [hunk 2]
```diff
@@ -181,4 +170,6 @@ __SYCL_ARCHITECTURE_ALIAS(intel_gpu_12_70_4, intel_gpu_mtl_u)
 __SYCL_ARCHITECTURE_ALIAS(intel_gpu_12_71_4, intel_gpu_mtl_h)
 __SYCL_ARCHITECTURE_ALIAS(intel_gpu_12_74_4, intel_gpu_arl_h)
 __SYCL_ARCHITECTURE_ALIAS(intel_gpu_20_1_4, intel_gpu_bmg_g21)
-__SYCL_ARCHITECTURE_ALIAS(intel_gpu_20_4_4, intel_gpu_lnl_m)
\ No newline at end of file
+__SYCL_ARCHITECTURE_ALIAS(intel_gpu_20_4_4, intel_gpu_lnl_m)
+__SYCL_ARCHITECTURE_ALIAS(intel_gpu_30_0_4, intel_gpu_ptl_h)
+__SYCL_ARCHITECTURE_ALIAS(intel_gpu_30_1_1, intel_gpu_ptl_u)
```
Comment (bader) [inline sycl/include/sycl/ext/oneapi/experimental/device_architecture.def:175]:
```suggestion
__SYCL_ARCHITECTURE_ALIAS(intel_gpu_30_1_1, intel_gpu_ptl_u)

```

**Review by bader — APPROVED** [review]
`sycl/doc/UsersManual.md` change looks good to me.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
@uditagarwal97, @intel/llvm-reviewers-runtime, could you please take a look?

File: llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td (hunk 1 above)
Comment (dm-vodopyanov) [reply llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td:192]: @jzc could you please add a doc and info about these new architectures in the separate upcoming PR?

**Review by steffenlarsen — APPROVED** [review]
LGTM! Given the urgency of this, could you please address @bader's comment in a follow-up? Comment doesn't seem to be a regression.

**Pull request merged**